silero_vad/silero_vad.py

```diff
@@ -317,31 +354,40 @@ def _multiprocessing_complete_tasks(
     """
     # Initialize and load the VAD:
     vad = VoiceActivityDetector(**vad_init_kwargs)
-    vad.load()
+    vad.load(force_reload=False)
 
     # Start listening to the tasks queue:
     while True:
         # Get the task:
-        task: BaseTask = tasks_queue.get()
+        task: Tuple[str, dict] = tasks_queue.get()
         if task == _MULTIPROCESSING_STOP_MARK:
             break
         try:
+            # Create the task:
+            task = TaskCreator.from_tuple(task_tuple=task)
             # Run the file through the VAD:
             speech_timestamps = vad.detect_voice(audio_file=task.audio_file)
             # Complete the task:
             task.do_task(speech_timestamps=speech_timestamps)
-            # Collect the result:
-            results_queue.put((False, task.get_result()))
+            # Build the result:
+            result = (False, task.get_result())
         except Exception as exception:
-            # Collect the error:
-            results_queue.put((True, (task.audio_file.name, str(exception))))
+            # Build the error:
+            result = (True, (task.audio_file.name, str(exception)))
+        # Collect the result / error:
+        results_queue.put(result)
 
     # Mark the end of the tasks:
     results_queue.put(_MULTIPROCESSING_STOP_MARK)
 
 
 # Get the global logger:
-_LOGGER = logging.getLogger()
+try:
+    import mlrun
+
+    _LOGGER = mlrun.get_or_create_ctx("silero_vad").logger
+except ModuleNotFoundError:
+    _LOGGER = logging.getLogger()
```

Comment:
I see that there are logs in the start of `diarize` and the `detect_voice` functions, that can be logged also by mlrun's logger, so maybe its better to figure out the logger also in an earlier stage.

Reply:
it happens regardless on import of the module even if it's written after some code that uses the `_LOGGER` as it is in functions. Also the `diarize` and the `detect_voice` functions are written after it.

Reply:
you right